websockets/client.py

```diff
@@ -35,9 +35,8 @@ def write_http_request(self, path, headers):
 
         """
         self.path = path
-        self.request_headers = email.message.Message()
-        for name, value in headers:
-            self.request_headers[name] = value
+        self.request_headers = http.client.HTTPMessage()
+        self.request_headers._headers = headers     # HACK
```

Comment:
I looked at CPython's code [here](https://github.com/python/cpython/blob/9648088e6ccd6d0cc04f450f55628fd8eda3784c/Lib/http/client.py#L163), and HTTPMessage doesn't really add much, and it's only reluctantly part of the public API, if that.

What about the idea of defining a `websockets.HTTPMessage(email.message.Message)` class?  At the least, this would let you DRY up the four occurrences of:

    self.response_headers._headers = headers    # HACK

and confine the hack to one spot.  You could also add a custom property / method on the class to eliminate the need for separate `self.raw_*_headers` attributes on the `WebSocketCommonProtocol` class.

Also, it might be worth having a couple unit tests to check that setting `_headers` directly will continue to work in future Pythons -- giving you the same result as going through `email.message.Message`'s (slightly slower) public API, etc.